### PR TITLE
Allow for SSH options to be overridden when bootstrapping. Needed for specific use cases

### PR DESCRIPTION
**Pull request by jasmingacic**
This PR allow for SSH options to be passed to bootstrap. This is needed for specific provider requirements where it is needed to pass `ssh.StrictHostChecksNo`

//CC @achilleasa 
Signed-off-by: root <jasmin.gacic@gmail.com>

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -75,8 +75,10 @@ github.com/agnivade/levenshtein v1.0.1/go.mod h1:CURSv5d9Uaml+FovSIICkLbAUZ9S4Rq
 github.com/ajstarks/svgo v0.0.0-20181006003313-6ce6a3bcf6cd h1:JdtityihAc6A+gVfYh6vGXfZQg+XOLyBvla/7NbXFCg=
 github.com/ajstarks/svgo v0.0.0-20181006003313-6ce6a3bcf6cd/go.mod h1:K08gAheRH3/J6wwsYMMT4xOr94bZjxIelGM0+d/wbFw=
 github.com/alecthomas/template v0.0.0-20160405071501-a0175ee3bccc/go.mod h1:LOuyumcjzFXgccqObfd/Ljyb9UuFJ6TxHnclSeseNhc=
+github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 h1:JYp7IbQjafoB+tBA3gMyHYHrpOtNuDiK/uB5uXxq5wM=
```
Comment (achilleasa) [inline go.sum:78]: I think you may want to run `go mod tidy`

File: environs/bootstrap.go [hunk 2]
```diff
@@ -80,6 +85,12 @@ type BootstrapParams struct {
 	Force bool
 }
 
+// InstanceSSHOptionsFunc is a function that given an instance config, returns
+// a HostSSHOptionsFunc that can generate SSH options for accessing an instance.
+type InstanceSSHOptionsFunc func(*instancecfg.InstanceConfig) SSHOptionsFunc
+
+type SSHOptionsFunc func(host string) (*ssh.Options, func(), error)
```
Comment (achilleasa) [inline environs/bootstrap.go:92]: You will need a comment for exported types.

**Comment by achilleasa** [comment]
@jasmingacic I have just located and fixed a bug (I will sending a patch your way later today) in the equinix branch that enables us to use the regular cloudinit setup as with the other providers without the need to override the ssh settings. 

This PR will not be needed anymore and can be closed.

**Comment by hpidcock** [comment]
@achilleasa can you please close this PR if you think it is no longer required.

**Pull request closed without merging**